internal/controllers/language_detect/language_detect.go

```diff
@@ -235,3 +237,58 @@ func (ld *LanguageDetect) isBatFileOrShellFile(lang string) bool {
 	return strings.EqualFold(lang, "Batchfile") ||
 		strings.EqualFold(lang, "Shell")
 }
+
+// This method was added because if o project analyzed is submodule is not possible run tool gitleaks,
+// then was added this logic for check if .git in cloned folder in .horusec is a file.
+// Case yes, we can find where is original git folder and replace in .horusec folder for run gitleaks tool
+// without problems.
+// Case is a folder we can go to next step without breaking changes.
+func (ld *LanguageDetect) copyGitFolderWhenIsSubmodule(directory, folderDstName string) error {
+	if ld.configs.EnableGitHistoryAnalysis {
+		isGitSubmodule, originalFolderPath := ld.returnGitFolderOriginalIfIsSubmodule(filepath.Join(directory, ".git"))
+		if isGitSubmodule {
+			logger.LogErrorWithLevel(messages.MsgErrorCopyProjectToHorusecAnalysis,
+				os.RemoveAll(filepath.Join(folderDstName, ".git")))
+			err := copy.Copy(filepath.Join(directory, originalFolderPath), filepath.Join(folderDstName, ".git"),
+				func(src string) bool { return false })
+			if err != nil {
+				logger.LogErrorWithLevel(messages.MsgErrorCopyProjectToHorusecAnalysis, err)
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+//nolint:funlen // lines is not necessary broken
+func (ld *LanguageDetect) returnGitFolderOriginalIfIsSubmodule(directory string) (bool, string) {
+	fileInfo, err := os.Stat(directory)
+	if err != nil {
+		logger.LogError(messages.MsgErrorCopyProjectToHorusecAnalysis, err)
+		return false, ""
+	}
+	if fileInfo.IsDir() {
+		return false, ""
+	}
+	fileContentBytes, err := os.ReadFile(directory)
+	if err != nil {
+		logger.LogError(messages.MsgErrorCopyProjectToHorusecAnalysis, err)
+		return false, ""
+	}
+	return ld.validateSubModuleContent(fileContentBytes)
+}
+
+func (ld *LanguageDetect) validateSubModuleContent(fileContentBytes []byte) (bool, string) {
+	var fileContent string
+	const PrefixSubModule = "gitdir: "
```

Comment:
What do you think about move this constant outside this function and make private?

Reply:
sounds good for me, fixed